File: Simple Linear Regression.py
#Linear Regression

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

Dataset = pd.read_csv("Salary_Data.csv")

X= Dataset.iloc[:,:-1].values
y= Dataset.iloc[:,1].values

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size= 1/3, random_state = 0)  

from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

y_pred= regressor.predict(X_test)

#Train
plt.scatter(X_train, y_train, color= 'red')
plt.plot(X_train, regressor.predict(X_train), color='blue')
plt.title('Salary vs Experience(Training Set)')
plt.xlabel('Years of Experience')
plt.ylabel('salary')
plt.show()

#Test
plt.scatter(X_test, y_test, color= 'red')
plt.plot(X_train, regressor.predict(X_train), color='blue')
plt.title('Salary vs Experience(Test Set)')
plt.xlabel('Years of Experience')
plt.ylabel('salary')
plt.show()

File: Simple Linear Regression.R
#Simple Linear Regression

dataset = read.csv('Salary_Data.csv')
#dataset = dataset[,2:3]

library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
train_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split ==  FALSE)

# training_set[,2:3] = scale(training_set[,2:3])
# test_set[,2:3] = scale(test_set[,2:3])

regressor = lm(formula = Salary ~ YearsExperience, data = training_set)

y_pred = predict(regressor, newdata = test_set)

ggplot() +
  geom_point(aes(x=training_set$YearsExperience, y=training_set$Salary),
             colour='red') +
  geom_line(aes(x=training_set$YearsExperience, y=predict(regressor, newdata = train_set)),
            colour='blue') +
  ggtitle('Salary vs Experience(Training set)')+
  xlab('Years of experience') +
  ylab('Salary')

ggplot() +
  geom_point(aes(x=test_set$YearsExperience, y=test_set$Salary),
             colour='red') +
  geom_line(aes(x=training_set$YearsExperience, y=predict(regressor, newdata = train_set)),
            colour='blue') +
  ggtitle('Salary vs Experience(Test set)')+
  xlab('Years of experience') +
  ylab('Salary')



  
